packages/components/src/icon-library.ts

```diff
@@ -194,4 +200,7 @@ const iconDict = {
   'warning-standard': fa(faExclamationTriangle),
 };
 
-export default iconDict;
+export type IconId = keyof typeof iconLibrary;
+
+// TODO: Improve types
+export default iconLibrary as Record<IconId, (props: any) => ReactNode>;
```

Comment:
`any` is not ideal, but I'm ok with it for now since it's confined to the `icon` function, which has types for its interface with consumers.